lib/puppet/defaults.rb

```diff
@@ -1259,16 +1267,16 @@ def self.valid_file_checksum_types
       :group => "service",
       :mode => "0660",
       :create => true,
+      :deprecated => :completely,
       :desc => "Where the puppet master web server saves its access log. This is
         only used when running a WEBrick puppet master. When puppet master is
         running under a Rack server like Passenger, that web server will have
         its own logging behavior."
     },
     :masterport => {
       :default    => 8140,
-      :desc       => "The port for puppet master traffic. For puppet master,
-      this is the port to listen on; for puppet agent, this is the port
-      to make requests on. Both applications use this setting to get the port.",
+      :desc       => "The port for puppet puppet agent to make requests on.
+      Only used by the puppet agent.",
```

Comment:
It's mainly puppet agent, but it also affects puppet subcommand making REST requests to the puppetserver, e.g. `puppet facts upload`. Not sure how best to word that... maybe something related to the `/puppet` route? Also it's more like the default port, as it can be overridden for specific endpoints, e.g. `report_port`.